content/en/docs/concepts/configuration/secret.md

```diff
@@ -61,1189 +116,1167 @@ merged into the `data` field. If a key appears in both the `data` and the
-for credentials used for TLS server and/or client. However, using the builtin Secret
-type helps ensure the consistency of Secret format in your project; the API server
-does verify if the required keys are provided in a Secret configuration.
+The output is similar to:
 
-When creating a TLS Secret using `kubectl`, you can use the `tls` subcommand
-as shown in the following example:
+```
+admin
+```
 
 ```shell
-kubectl create secret tls my-tls-secret \
-  --cert=path/to/cert/file \
-  --key=path/to/key/file
+cat /etc/foo/password
 ```
 
-The public/private key pair must exist before hand. The public key certificate
-for `--cert` must be .PEM encoded (Base64-encoded DER format), and match the
-given private key for `--key`.
-The private key must be in what is commonly called PEM private key format,
-unencrypted. In both cases, the initial and the last lines from PEM (for
-example, `--------BEGIN CERTIFICATE-----` and `-------END CERTIFICATE----` for
-a cetificate) are *not* included.
+The output is similar to:
 
-### Bootstrap token Secrets
+```
+1f2d1e2e67df
+```
 
-A bootstrap token Secret can be created by explicitly specifying the Secret
-`type` to `bootstrap.kubernetes.io/token`. This type of Secret is designed for
-tokens used during the node bootstrap process. It stores tokens used to sign
-well known ConfigMaps.
+The program in a container is responsible for reading the secret data from these
+files, as needed.
 
-A bootstrap token Secret is usually created in the `kube-system` namespace and
-named in the form `bootstrap-token-<token-id>` where `<token-id>` is a 6 character
-string of the token ID.
+#### Mounted Secrets are updated automatically
 
-As a Kubernetes manifest, a bootstrap token Secret might look like the
-following:
+When a volume contains data from a Secret, and that Secret is updated, Kubernetes tracks
+this and updates the data in the volume, using an eventually-consistent approach.
+
+{{< note >}}
+A container using a Secret as a
+[subPath](/docs/concepts/storage/volumes#using-subpath) volume mount does not receive
+automated Secret updates.
+{{< /note >}}
+
+The kubelet keeps a cache of the current keys and values for the Secrets that are used in
+volumes for pods on that node.
+The type of the cache is configurable using the `ConfigMapAndSecretChangeDetectionStrategy` field in
+the [KubeletConfiguration struct](https://github.com/kubernetes/kubernetes/blob/{{< param "docsbranch" >}}/staging/src/k8s.io/kubelet/config/v1beta1/types.go).
+
+Updates to Secrets can be either propagated by an API watch mechanism (the default), based on
+a cache with a defined time-to-live, or polled from the cluster API server on each kubelet
+synchronisation loop.
+
+As a result, the total delay from the moment when the Secret is updated to the moment
+when new keys are projected to the Pod can be as long as the kubelet sync period + cache
+propagation delay, where the cache propagation delay depends on the chosen cache type
```

Comment:
nit: This paragraph should be broken up into several sentences (or at least 2).
As a result, the total delay ...